pkg/watchdog/watchdog.go

```diff
@@ -0,0 +1,236 @@
+func (w *Watchdog) Close() error {
+	return w.Close()
+}
+
+// MagicClose disarms the watchdog.
+func (w *Watchdog) MagicClose() error {
+	if _, err := w.f.Write([]byte("V")); err != nil {
+		return err
+	}
+	return w.Close()
+}
+
+// GetSupport returns the WatchdogInfo struct.
+func (w *Watchdog) GetSupport() (WatchdogInfo, error) {
+	var wi WatchdogInfo
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocGetSupport, uintptr(unsafe.Pointer(&wi))); err != 0 {
+		return wi, err
+	}
+	return wi, nil
+}
+
+// GetStatus returns the current status.
+func (w *Watchdog) GetStatus() (int32, error) {
+	var flags int32
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocGetStatus, uintptr(unsafe.Pointer(&flags))); err != 0 {
+		return StatusUnknown, err
+	}
+	return flags, nil
+}
+
+// GetBootStatus returns the status at the last reboot.
+func (w *Watchdog) GetBootStatus() (int32, error) {
+	var flags int32
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocGetBootStatus, uintptr(unsafe.Pointer(&flags))); err != 0 {
+		return StatusUnknown, err
+	}
+	return flags, nil
+}
+
+// GetTemp returns temperature in degrees Fahrenheit.
+func (w *Watchdog) GetTemp() (int32, error) {
+	var temp int32
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocGetTemp, uintptr(unsafe.Pointer(&temp))); err != 0 {
+		return 0, err
+	}
+	return temp, nil
+}
+
+// SetOptions can be used to control some aspects of the cards operation.
+func (w *Watchdog) SetOptions(options int32) error {
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocSetOptions, uintptr(unsafe.Pointer(&options))); err != 0 {
+		return err
+	}
+	return nil
+}
+
+// KeepAlive pets the watchdog.
+func (w *Watchdog) KeepAlive() error {
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocKeepAlive, 0); err != 0 {
+		return err
+	}
+	return nil
+}
+
+// SetTimeout sets the watchdog timeout on the fly. It returns the real timeout
+// used and this timeout might differe from the requested one due to limitation
+// of the hardware.
+func (w *Watchdog) SetTimeout(timeout int32) (int32, error) {
+	originalTimeout := timeout
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocSetTimeout, uintptr(unsafe.Pointer(&timeout))); err != 0 {
+		return 0, err
+	}
+	if originalTimeout != timeout {
+		log.Printf("Watchdog timeout set to %ds, wanted %ds", timeout, originalTimeout)
+	}
+	return timeout, nil
+}
+
+// GetTimeout returns the current watchdog timeout.
+func (w *Watchdog) GetTimeout() (int32, error) {
+	var timeout int32
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocGetTimeout, uintptr(unsafe.Pointer(&timeout))); err != 0 {
+		return 0, err
+	}
+	return timeout, nil
+}
+
+// SetPreTimeout sets the watchdog pretimeout on the fly. The pretimeout is the
+// number of seconds before the timeout before triggering the preaction (such
+// as an NMI, interrupt, ...)
+func (w *Watchdog) SetPreTimeout(timeout int32) (int32, error) {
```

Comment:
time.Duration

Reply:
See the other comment about this always being in seconds and this removes the need for rounding.

Reply:
I don't think the rounding is that hurtful, but you could return an error if it doesn't cleanly divide.